core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/BlueprintUtilV4Controller.java

```diff
@@ -49,7 +53,8 @@
      */
     @Override
     @Deprecated
-    public RecommendationV4Response createRecommendation(Long workspaceId, String blueprintName, String credentialName,
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.DESCRIBE_CREDENTIAL)
+    public RecommendationV4Response createRecommendation(Long workspaceId, String blueprintName, @ResourceName String credentialName,
```

Comment:
Shouldn't we authorize on blueprintName as well? (I know the framework doesn't allow it right now.)

Reply:
yes, i couldn't decide which one to authorize against, maybe credential is more sensitive, so i have chosen  credential, it is still better than leave it unathorized. and as you said, framework is unable to authorize both, and the task is not aiming to solve this, so currently this is the best i can do here

Reply:
Yes, I agree.